sdk/core/Azure.Core/src/Pipeline/HttpClientTransportOptions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Azure.Core.Pipeline
+{
+    /// <summary>
+    /// Enables configuration of options for the <see cref="HttpClientTransport"/>
+    /// </summary>
+    public class HttpClientTransportOptions
+    {
+        /// <summary>
+        /// A callback that defines whether to validate the certificate presented by the server.
+        /// </summary>
+        public Func<X509Certificate2?, bool>? ServerCertificateCustomValidationCallback { get; set; }
+
+        /// <summary>
+        /// If <c>true</c>, all service certificates are treated as trusted.
+        /// </summary>
+        public bool TrustAllServerCertificates { get; set; }
```

Comment:
Should this be `TrustAllCertificatesWhenCallbackNotSupported`?

Reply:
I like the idea of keeping it generic as a way to simply trust all certs. Callback not being supported is just one reason why someone might set this.

Reply:
How would they know when to set it? I'm afraid everyone would just set this to true because it's too hard to know when it's needed.

Reply:
We could expand on these edge cases in the doc comments and/or add additional context to the exception message if the setter throws.

Reply:
Would every library then have to implement detection for them?

Reply:
Would custom detection be necessary? The error related to trying to set the callback would be immediately encountered rather than situational at run time.

Reply:
yeah, so every library would have to have something like:

```
if (SomehowDetectMacOS() || some other condition that libraries won't immediately discover)
{
	options.TrustAllServerCertificates  = true;
}
else
{
	options.ServerCertificateCustomValidationCallback = _ => ...;
}
```

Reply:
I don't think a property named `TrustAllCertificatesWhenCallbackNotSupported` avoids this - especially for scenarios where the callback error is not immediately discoverable.

Even for this scenario on Mac, there are apparently ways to make it work. 
https://github.com/dotnet/runtime/issues/17723#issuecomment-234418291

If the BCL doesn't special case this callback I'm not convinced we should.

Reply:
Nice thread. 

My general goal here is for clients to have as little logic as possible and for Core to do all it can to "just make things work".

Reply:
I'm in total agreement with the goal - I'm just not sure what would be clearly a better approach. Perhaps we eliminate this property totally and only add one if we get feedback.

Reply:
won't your client be broken by-default on mac if we don't do anything here?

Reply:
This reminds me - the CL client needs an option to disable to automatic cert trust initialization. But with that option, either we enable it by default and ensure the way to disable it is discoverable, or disable make it easy to find how to get the default trust experience.

Reply:
I might be wrong and the cert validation might be fixed by now (in 3.1). We should try.

Reply:
I tested this on Mac and it looks like setting the callback works fine on 3.1, which is the current minimum version. Looks like it is safe to remove this.